### PR TITLE
'main': Fix highlighting of «<->»-style numeric globs.

### PR DESCRIPTION
**Pull request by danielshahaf**

### CODE REVIEW

**Comment by phy1729** [comment]
This highlights `print 0<->` as a redirection.

**Comment by danielshahaf** [comment]
... instead of as a glob pattern for "the character `0` followed by a number".

**Comment by danielshahaf** [comment]
35270f7c97b9 ?

File: highlighters/main/test-data/number_range-glob.zsh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+# IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+# OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# -------------------------------------------------------------------------------------------------
+# -*- mode: zsh; sh-indentation: 2; indent-tabs-mode: nil; sh-basic-offset: 2; -*-
+# vim: ft=zsh sw=2 ts=2 et
+# -------------------------------------------------------------------------------------------------
+
+BUFFER=$'print <-> 0<->'
```
Comment (phy1729) [inline highlighters/main/test-data/number_range-glob.zsh]: Perhaps add a `<foo->` which is two redirections.

**Comment by phy1729** [comment]
Would incorrectly match `foo<bar-baz>quux`, but `(z)` saves us by splitting the redirects into separate words. Needs a rebase post arg-quoting merge.

File: highlighters/main/test-data/number_range-glob.zsh (hunk 1 above)
Comment (danielshahaf) [reply highlighters/main/test-data/number_range-glob.zsh]: Added.

**Comment by danielshahaf** [comment]
> Would incorrectly match foo<bar-baz>quux, but (z) saves us by splitting the redirects into separate words.

So, no changes needed, right?

> Needs a rebase post arg-quoting merge.

Yeah, and a squash.  Will do.

**Comment by phy1729** [comment]
> So, no changes needed, right?
Yep. Just the rebase.

**Comment by danielshahaf** [comment]
The rebase is blocked by #468.

**Comment by danielshahaf** [comment]
The rebase is not trivial, the logic moved to a characterwise loop in `_zsh_highlight_main_highlighter_highlight_argument`.

That doesn't get called on redirections (https://github.com/zsh-users/zsh-syntax-highlighting/blob/df0f44858634c454fbbabc182fc5ffe43b8907b5/highlighters/main/main-highlighter.zsh#L695-L698) so any `<` not quoted (by backslash or quotes) should be part of the `<->` globbing syntax.

**Comment by danielshahaf** [comment]
Okay, how about https://github.com/zsh-users/zsh-syntax-highlighting/compare/master...danielshahaf:numeric-globs-v2 (0decb2a8f541)?

**Comment by danielshahaf** [comment]
Merged after @phy1729 review.  (The PR is "closed"/red rather than "merged"/purple because it's about -v1 and it's -v2 that got merged.)

**Comment by danielshahaf** [comment]
The commit is 987b743646c3e71a160de0ac2177bfe914bde39a in master.

**Pull request closed without merging**